### PR TITLE
nodejs*: update

### PR DESCRIPTION
**Pull request by k4leg**
Do I need to rebuild packages that depend on `nodejs*` or `nodejs*-devel`?
(https://nodejs.org/en/blog/vulnerability/december-2019-security-releases/)

### CODE REVIEW

**Comment by k4leg** [comment]
Hmm ... I don’t understand how to solve this [problem](http://dpaste.com/2WMN5Z9). I am new and in such beams do I need to contact [here](https://github.com/nodejs/node/issues) to solve this problem?

File: srcpkgs/nodejs-lts-10/template [hunk 1]
```diff
@@ -31,38 +31,42 @@ replaces="iojs>=0"
 conflicts="nodejs nodejs-lts"
 provides="nodejs-runtime-0_1"
 
-if [ "$XBPS_WORDSIZE" -ne "$XBPS_TARGET_WORDSIZE" ]; then
-	nocross="host and target must have the same pointer size"
+if [ ! "$XBPS_WORDSIZE" == "$XBPS_TARGET_WORDSIZE" ]; then
```
Comment (xtraeme) [inline srcpkgs/nodejs-lts-10/template]: please replace this with a simple `"$XBPS_WORDSIZE" != "$XBPS_TARGET_WORDSIZE"``

File: srcpkgs/nodejs-lts-10/template [hunk 2]
```diff
@@ -31,38 +31,42 @@ replaces="iojs>=0"
...
+	nocross="Host and target must have the same pointer size."
 fi
 
 case "$XBPS_MACHINE" in
-	ppc64*) ;;
-	mips*|ppc*) hostmakedepends+=" libatomic-devel" ;;
-	*) ;;
+	'mips*'|'ppc'|'ppc-musl') hostmakedepends+=" libatomic-devel";;
```
Comment (xtraeme) [inline srcpkgs/nodejs-lts-10/template]: no need to quote them

File: srcpkgs/nodejs-lts-10/template [hunk 3]
```diff
@@ -31,38 +31,42 @@ replaces="iojs>=0"
...
-			i686*) _args="--dest-cpu=x86" ;;
-			x86_64*) _args="--dest-cpu=x86_64" ;;
-			*) msg_error "$pkgver: cannot be cross compiled for ${XBPS_TARGET_MACHINE}.\n" ;;
+			'arm*') _args+=" --dest-cpu=arm";;
+			'aarch64*') _args+=" --dest-cpu=arm64";;
+			'ppc64*') _args+=" --dest-cpu=ppc64";;
+			'ppc*') _args+=" --dest-cpu=ppc";;
+			'mipsel*') _args+=" --dest-cpu=mipsel";;
+			'mips*') _args+=" --dest-cpu=mips";;
+			'i686*') _args+=" --dest-cpu=x86";;
```
Comment (xtraeme) [inline srcpkgs/nodejs-lts-10/template]: again, quoting is unnecessary

File: srcpkgs/nodejs-lts-10/template [hunk 4]
```diff
@@ -31,38 +31,42 @@ replaces="iojs>=0"
...
+			'x86_64*') _args+=" --dest-cpu=x86_64";;
+			'*') msg_error "${pkgname}-${pkgver}: cannot be cross compiled for ${XBPS_TARGET_MACHINE}.";;
```
Comment (xtraeme) [inline srcpkgs/nodejs-lts-10/template]: `pkgver` == `${pkgname}-${version}_${revision}`

File: srcpkgs/nodejs-lts-10/template (hunk 1 above)
Comment (q66) [reply srcpkgs/nodejs-lts-10/template]: why replace it at all? the previous one is fine

File: srcpkgs/nodejs-lts-10/template (hunk 2 above)
Comment (q66) [reply srcpkgs/nodejs-lts-10/template]: again this change is pointless, it was fine as it was

**Comment by q66** [comment]
pretty much anything involving formatting and arch conditionals in this needs to be discarded

File: srcpkgs/nodejs-lts-10/template [hunk 5]
```diff
@@ -31,38 +31,42 @@ replaces="iojs>=0"
...
 		esac
-		_args+=" --without-snapshot"
+		_args+=" --without-snapshot --cross-compiling"
+	else
+		_args+=" --no-cross-compiling"
```
Comment (q66) [inline srcpkgs/nodejs-lts-10/template]: redundant

File: srcpkgs/nodejs/template [hunk 6]
```diff
@@ -32,42 +31,36 @@ replaces="iojs>=0"
...
+	'mips*'|'ppc64*') makedepends+=" libatomic-devel";;
+	'ppc*') broken="Node ${version} does not support 32-bit ppc.";;
 esac
 
 do_configure() {
 	local _args
 
+	_args="--dest-os=linux --use-largepages"
 	export LD="$CXX"
+	export CFLAGS="-Wreinterpret_cast -Wmaybe-uninitialized"
```
Comment (q66) [inline srcpkgs/nodejs/template]: ???

**Comment by q66** [comment]
the `llhttp_set_lenient` error should be fixed by updating `llhttp` first

File: srcpkgs/nodejs-lts-10/template (hunk 1 above)
Comment (xtraeme) [reply srcpkgs/nodejs-lts-10/template]: yeah, indeed. Didn't see it

**Comment by k4leg** [comment]
@q66 Thanks for the help! Updating `llhttp` to the latest version fixed the problem.

File: srcpkgs/nodejs-lts-10/template [hunk 7]
```diff
@@ -32,37 +32,39 @@ conflicts="nodejs nodejs-lts"
...
-			*) msg_error "$pkgver: cannot be cross compiled for ${XBPS_TARGET_MACHINE}.\n" ;;
+			'arm*') _args=" --dest-cpu=arm";;
+			'aarch64*') _args=" --dest-cpu=arm64";;
+			'ppc64*') _args=" --dest-cpu=ppc64";;
+			'ppc*') _args=" --dest-cpu=ppc";;
+			'mipsel*') _args=" --dest-cpu=mipsel";;
+			'mips*') _args=" --dest-cpu=mips";;
+			'i686*') _args=" --dest-cpu=x86";;
+			'x86_64*') _args=" --dest-cpu=x86_64";;
+			'*') msg_error "${pkgver}: cannot be cross compiled for ${XBPS_TARGET_MACHINE}.\n";;
```
Comment (xtraeme) [inline srcpkgs/nodejs-lts-10/template]: The quoting is unnecessary

File: srcpkgs/nodejs-lts-10/template (hunk 2 above)
Comment (xtraeme) [reply srcpkgs/nodejs-lts-10/template]: Not resolved, unnecessary quoting

**Comment by k4leg** [comment]
It’s better read, isn't it?

**Comment by q66** [comment]
no, it's not, and even if it was, it wouldn't belong in the PR

**Comment by q66** [comment]
there are whitespace issues in the templates, fix them

**Comment by xtraeme** [comment]
Seriously, do it OK or I will give up. This is not a game.

**Comment by xtraeme** [comment]
We told you 3 or more times, your quoting is wrong and stupid. So? Have a nice day.

**Comment by q66** [comment]
hmm i don't see any more quotes

**Pull request closed without merging**